engine/src/main/java/org/terasology/rendering/nui/layers/mainMenu/StartPlayingScreen.java

```diff
@@ -16,18 +16,93 @@
 package org.terasology.rendering.nui.layers.mainMenu;
 
 import org.terasology.assets.ResourceUrn;
+import org.terasology.config.Config;
+import org.terasology.context.Context;
+import org.terasology.engine.GameEngine;
+import org.terasology.engine.SimpleUri;
+import org.terasology.engine.TerasologyConstants;
+import org.terasology.engine.modes.StateLoading;
+import org.terasology.engine.module.ModuleManager;
+import org.terasology.game.GameManifest;
+import org.terasology.module.DependencyResolver;
+import org.terasology.module.Module;
+import org.terasology.module.ResolutionResult;
+import org.terasology.network.NetworkMode;
+import org.terasology.registry.In;
+import org.terasology.rendering.assets.texture.Texture;
 import org.terasology.rendering.nui.CoreScreenLayer;
 import org.terasology.rendering.nui.WidgetUtil;
+import org.terasology.rendering.nui.widgets.UIImage;
+import org.terasology.rendering.world.WorldSetupWrapper;
+import org.terasology.world.internal.WorldInfo;
+import org.terasology.world.time.WorldTime;
 
 public class StartPlayingScreen extends CoreScreenLayer {
 
+    @In
+    private ModuleManager moduleManager;
+
+    @In
+    private Config config;
+
+    @In
+    private GameEngine gameEngine;
+
     public static final ResourceUrn ASSET_URI = new ResourceUrn("engine:startPlayingScreen");
+    private Texture texture;
+    private WorldSetupWrapper world;
+    private UniverseWrapper universeWrapper;
 
     @Override
     public void initialise() {
 
         WidgetUtil.trySubscribe(this, "close", button ->
                 triggerBackAnimation()
         );
+
+        WidgetUtil.trySubscribe(this, "play", button -> {
+            GameManifest gameManifest = new GameManifest();
+
+            gameManifest.setTitle(universeWrapper.getGameName());
+
+            DependencyResolver resolver = new DependencyResolver(moduleManager.getRegistry());
+            ResolutionResult result = resolver.resolve(config.getDefaultModSelection().listModules());
+            if (!result.isSuccess()) {
+                MessagePopup errorMessagePopup = getManager().pushScreen(MessagePopup.ASSET_URI, MessagePopup.class);
+                if (errorMessagePopup != null) {
+                    errorMessagePopup.setMessage("Invalid Module Selection", "Please review your module seleciton and try again");
+                }
+                return;
+            }
+            for (Module module : result.getModules()) {
+                gameManifest.addModule(module.getId(), module.getVersion());
+            }
+
+            SimpleUri uri = world.getWorldGeneratorInfo().getUri();
+            float timeOffset = 0.25f + 0.025f;
```

Comment:
Magic numbers? Might be an idea to extract info a constant that could also explain what the number means